ui.frontend/src/view/InstanceManager.js

```diff
@@ -160,18 +160,19 @@ export default class InstanceManager {
         //get the model from added instance state
         let addedModel = this.formContainer.getModel(addedInstanceJson.id);
         this.updateCloneIds(htmlElement, 'temp_0', addedModel);
-
-        //no child exist in the view
-        if (this.children.length == 0) {
-            this.parentElement.append(htmlElement);
-            //this.markerElement.after(htmlElement);
-        } else if (instanceIndex == 0) {
-            //special case for first element
-            let afterElement = this.children[0].element.parentElement;
-            this.parentElement.insertBefore(htmlElement, afterElement);
+        if (this.repeatableParentView && (typeof this.repeatableParentView.customRepeatableHtmlHandler === "function")) {
+            this.repeatableParentView.customRepeatableHtmlHandler(this, addedModel, htmlElement, beforeElement);
         } else {
-            let beforeViewElement = (beforeElement != null) ? beforeElement : this.children[instanceIndex - 1].element.parentElement;
-            beforeViewElement.after(htmlElement);
+            // this is required if repeatableParentView is the formContainer.
+            if (this.children.length == 0) {
+                this.parentElement.append(htmlElement);
+            } else if (addedModel.index == 0) {
+                let afterElement = this.children[0].element.parentElement;
+                this.parentElement.insertBefore(htmlElement, afterElement);
+            } else {
+                let beforeViewElement = (beforeElement != null) ? beforeElement : this.children[instanceIndex - 1].element.parentElement;
+                beforeViewElement.after(htmlElement);
```

Comment:
you have repeated this logic in customRepeatableHandler, encapsulate this in function and invoke from both places.

Reply:
Removed this from FormPanel as the same logic is required for PanelContainer as well as FormContainer, both of them are handled by instanceManager now.